### ISSUE TITLE
Add and remove to favorites

### ISSUE THREAD

**Issue by Valdnet** [nextcloud/talk-android#1270]
## Steps to reproduce

1. Create a conversation.
2. Add conversation to favorites.
3. Delete conversation as favorites.

The application **shuts down** when adding or removing a conversation to favorites.

## Device information

**Android version:** 10

**Talk version:** 12.0.0 Alpha 13

## Server information

**Nextcloud version:** 21.0.1

**Talk version:** 11.2.1

**Comment by AndyScherzinger** [comment]
```
Caused by: java.lang.ClassCastException: com.nextcloud.talk.models.json.generic.GenericOverall cannot be cast to com.nextcloud.talk.models.json.conversations.RoomOverall
        at com.nextcloud.talk.controllers.bottomsheet.OperationsMenuController$OperationsObserver.onNext(OperationsMenuController.java:703)
        at io.reactivex.internal.operators.observable.ObservableRetryPredicate$RepeatObserver.onNext(ObservableRetryPredicate.java:69)
        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
        at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:124)
        at android.os.Handler.handleCallback(Handler.java:938) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:233) 
        at android.app.ActivityThread.main(ActivityThread.java:8035) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978) 
```

**Comment by AndyScherzinger** [comment]
@mahibi @nickvergessen adding/removing from favorites uses:
```
ncApi.removeConversationFromFavorites
ncApi.addConversationToFavorites
```
So the observer mechanism provides a `GenericOverall` object since that is the result of the above calls. anfortunately this then leads to the stack trace since the code tries to cast it to a `RoomOverall` object which it can't and thus crashes. Either the ncApi implementation has been changed or this has never worked (?)

**Comment by AndyScherzinger** [comment]
Issue introduced right here: https://github.com/nextcloud/talk-android/commit/213d11358addc6a0fecd46efd99a6729060783cb#diff-e32e7dc27557fad2f2c558b814e01d3f2080fba6d0eadd48c0745f55d8108429R700-R701

**Comment by AndyScherzinger** [comment]
Fix is in #1276 - please review @nickvergessen 😃
